reactioncommerce/reaction
# WIP feat: Introduce Mobx for global state management - admin sidebar example

Resolves #4580 
Impact: **minor**  
Type: **feature**

## Solution
This PR adds `mobx`, creates a UI store, and converts several session variables used in the admin sidebar, to mobx observables.

In order to keep the variables reactive, `autorun` from Mobx is used. This example shows its use in both Blaze and React (containers). 

## Testing
1. Login as admin
2. Confirm the admin sidebar still functions correctly. Be sure to open and close the order detail view during testing.
3. Visit the grid page
4. Click through to PDP
5. Click the various edit buttons (in edit mode) and confirm the sidebar behaves appropriately
6. Click back to the main product grid
7. Confirm no console errors
